app/api/users/[id]/route.ts

```
@@ -0,0 +1,42 @@
+import pool from '@/lib/db';
+
+export async function GET(
+    _req: Request,
+    { params }: { params: { id: string } }
+) {
+    try {
+        const id = params.id;
+        
+        if (!id || !/^[0-9]+$/.test(id)) {
+            return new Response(JSON.stringify({ error: 'Invalid user ID format' }), { status: 400 });
```

You can throw an error and handle the error in the catch block:
```js
try{
            ...
            if (!id || !/^[0-9]+$/.test(id)) {
                        throw new Error('invalid_id');
            ...
} catch(error) {
            if (error instanceof Error && error.message === 'invalid_id') {
                        return Response.json({ error: 'Invalid user ID format' }, { status: 400 });
            }
            if (error instanceof Error) {
                        return Response.json({ error: error.message }, { status: 500 });
            }
            return Response.json({ error: error }, { status: 500 });
}
```